lib/client-assets.php

```diff
@@ -799,6 +799,41 @@ function gutenberg_capture_code_editor_settings( $settings ) {
 	$gutenberg_captured_code_editor_settings = $settings;
 	return false;
 }
+/**
+ * Retrieve a stored autosave that is newer than the post save.
+ *
+ * Deletes autosaves that are older than the post save.
+ *
+ * @param  WP_Post $post Post object.
+ * @return WP_Post|boolean The post autosave. False if none found.
+ */
+function get_autosave_newer_than_post_save( $post ) {
+	// Add autosave data if it is newer and changed.
+	$autosave      = wp_get_post_autosave( $post->ID );
+	$show_autosave = false;
+
+	// Check if we have an autosave that is newer than the post and different from the current post.
+	if (
+		$autosave &&
+		mysql2date( 'U', $autosave->post_modified_gmt, false ) > mysql2date( 'U', $post->post_modified_gmt, false )
+	) {
+		// Iterate thru revisioned fields checking for any changes.
+		foreach ( _wp_post_revision_fields( $post ) as $autosave_field => $_autosave_field ) {
+			if ( normalize_whitespace( $autosave->$autosave_field ) != normalize_whitespace( $post->$autosave_field ) ) {
```

Comment:
This is not needed any more. I know it exists in edit-form-advanced.php but it is here to catch identical autosaves when editing super old posts (from before 2.7 I think).

I'd be +1 to skip it here.

Reply:
do you mean the entire check for changed revisioned fields or just the normalize_whitespace part?

Reply:
Uh, sorry for not being clear. Yes, the entire check. It was intended to fix existing posts that used to have identical autosave revisions at some point, very long tie ago :)